EIPS/eip-6960.md

```diff
@@ -0,0 +1,328 @@
+---
+eip: 6960
+title: Dual Layer Token
+description: Token with a two-level classification system using mainId and subId
+author: Adam Boudjemaa (@aboudjem), Mohamad Hammoud (@mohamadhammoud), Nawar Hisso (@nawar-hisso), Khawla Hassan (@khawlahssn), Mohammad Zakeri Rad (@zakrad), Ashish Sood <soodgen@gmail.com>
+discussions-to: https://ethereum-magicians.org/t/eip-6960-dual-layer-token/14070
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-04-30
+---
+
+## Abstract
+
+The dual-layer token combines the functionalities of [ERC-20](./eip-20.md), [ERC-721](./eip-721.md), and [ERC-1155](./eip-1155.md) while adding a classification layer that uses `mainId` as the main asset type identifier and `subId` as the unique attributes or variations of the main asset.
+![Dual Layer Token](../assets/eip-6960/eip-6960-dual-layer-token-dlt.png)
+
+The proposed token aims to offer more granularity in token management, facilitating a well-organized token ecosystem and simplifying the process of tracking tokens within a contract. This standard is particularly useful for tokenizing and enabling the fractional ownership of Real World Assets (RWAs). It also allows for efficient and flexible management of both fungible and non-fungible assets.
+
+The following are examples of assets that the DLT standard can represent fractional ownership of:
+
+- Invoices
+- Company stocks
+- Digital collectibles
+- Real estate
+
+## Motivation
+
+The [ERC-1155](./eip-1155.md) standard has experienced considerable adoption within the Ethereum ecosystem; however, its design exhibits constraints when handling tokens with multiple classifications, particularly in relation to Real World Assets (RWAs) and fractionalization of assets.
+
+This EIP strives to overcome this limitation by proposing a token standard incorporating a dual-layer classification system, allowing for enhanced organization and management of tokens, especially in situations where additional sub-categorization of token types is necessary.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### DLT Interface
+
+```solidity
+// SPDX-License-Identifier: MIT
```

Comment:
Code inline in the proposal must be licensed under `CC0-1.0`. Code in the assets directory can be MIT if you'd like.